core/src/main/java/org/apache/iceberg/mapping/MappingUtil.java

```diff
@@ -302,6 +302,11 @@ public MappedFields map(Types.MapType map, MappedFields keyResult, MappedFields
           MappedField.of(map.valueId(), "value", valueResult));
     }
 
+    @Override
+    public MappedFields variant(Types.VariantType variant) {
+      return null; // no mapping because variant no nested fields
```

Comment:
Minor: "no mapping because variant has no nested fields with IDs"

It is missing "has" (typo) and I think it is good to state that there are no nested IDs because there can be nested data within a variant.